travis fails to build Rasterio

`rasterio/_base.c:287:22: fatal error: cpl_conv.h: No such file or directory` - looks like no python-dev. In fact, looks like this repo is set up to use old linux wheels.

@sgillies maybe this would be a good place to employ the manylinux wheels for 1.0a1?

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,7 +18,7 @@ cache:
     - $HOME/.pip-cache/
 install:
   - "pip install -U pip"
-  - "pip install numpy>=1.10"
+  - "pip install numpy>=1.10 -vv"
   - "pip install --pre rasterio==1.0a1"
   - "pip list"
   - "pip install -e .[test]"

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,7 +20,7 @@ cache:
 install:
   - pip install -U pip
   - pip install "numpy>=1.10"
-  - pip install --no-binary rasterio rasterio==1.0a1
+  - pip install --no-binary --pre rasterio rasterio==1.0a1
   - pip install -e .[test]
 script:
   - py.test --cov mbtiles --cov-report term-missing

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,7 +19,7 @@ cache:
 install:
   - "pip install -U pip"
   - "pip list"
-  - "pip install numpy>=1.10"
+  - pip install "numpy>=1.10"
   - "pip install --pre rasterio==1.0a1"
   - "pip list"
   - "pip install -e .[test]"

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,17 +12,15 @@ addons:
       - gdal-bin
 python:
   - "2.7"
-  - "3.4"
+  - "3.5"
 cache:
   directories:
     - $HOME/.pip-cache/
 install:
-  - "pip install -U pip"
-  - "pip list"
+  - pip install -U pip
   - pip install "numpy>=1.10"
-  - "pip install --pre rasterio==1.0a1"
-  - "pip list"
-  - "pip install -e .[test]"
+  - pip install --no-binary rasterio rasterio==1.0a1
+  - pip install -e .[test]
 script:
   - py.test --cov mbtiles --cov-report term-missing
 after_success:

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,6 +19,7 @@ cache:
     - $HOME/.pip-cache/
 install:
   - pip install -U pip
+  - python -c "import sys; print(sys.maxunicode)"
   - pip install "numpy>=1.10"
   - pip install --pre --no-binary rasterio rasterio==1.0a1
   - pip install -e .[test]

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,7 +20,7 @@ cache:
 install:
   - pip install -U pip
   - pip install "numpy>=1.10"
-  - pip install --no-binary --pre rasterio rasterio==1.0a1
+  - pip install --pre --no-binary rasterio rasterio==1.0a1
   - pip install -e .[test]
 script:
   - py.test --cov mbtiles --cov-report term-missing

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,8 +19,6 @@ cache:
     - $HOME/.pip-cache/
 install:
   - pip install -U pip
-  - python -c "import sys; print(sys.maxunicode)"
-  - pip install "numpy>=1.10"
   - pip install --pre --no-binary rasterio rasterio==1.0a1
   - pip install -e .[test]
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,6 @@
 sudo: false
 language: python
+dist: trusty
 env:
   global:
     - PIP_WHEEL_DIR=$HOME/.pip-cache/
@@ -10,6 +11,7 @@ addons:
       - libgdal1h
       - gdal-bin
 python:
+  - "2.7"
   - "3.4"
 cache:
   directories:
@@ -18,6 +20,7 @@ install:
   - "pip install -U pip"
   - "pip install numpy>=1.10"
   - "pip install --pre rasterio==1.0a1"
+  - "pip list"
   - "pip install -e .[test]"
 script:
   - py.test --cov mbtiles --cov-report term-missing

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,13 +10,13 @@ addons:
       - libgdal1h
       - gdal-bin
 python:
-  - "2.7"
   - "3.4"
 cache:
   directories:
     - $HOME/.pip-cache/
 install:
   - "pip install -U pip"
+  - "pip install numpy>=1.10"
   - "pip install --pre rasterio==1.0a1"
   - "pip install -e .[test]"
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,9 @@
 sudo: false
 language: python
 env:
-  - RASTERIO_VERSION=0.26.0
+  global:
+    - PIP_WHEEL_DIR=$HOME/.pip-cache/
+    - PIP_FIND_LINKS=$HOME/.pip-cache/
 addons:
   apt:
     packages:
@@ -13,12 +15,10 @@ python:
 cache:
   directories:
     - $HOME/.pip-cache/
-    - $HOME/wheelhouse
-before_install:
-  - "./.travis_riodeps.sh"
 install:
   - "pip install -U pip"
-  - "pip install --use-wheel --find-links=$HOME/wheelhouse -e .[test] --cache-dir $HOME/.pip-cache"
+  - "pip install --pre rasterio==1.0a1"
+  - "pip install -e .[test]"
 script:
   - py.test --cov mbtiles --cov-report term-missing
 after_success:

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,8 +3,8 @@ language: python
 dist: trusty
 env:
   global:
-    - PIP_WHEEL_DIR=$HOME/.pip-cache/
-    - PIP_FIND_LINKS=$HOME/.pip-cache/
+    - PIP_WHEEL_DIR=$HOME/.cache/pip
+    - PIP_FIND_LINKS=$HOME/.cache/pip
 addons:
   apt:
     packages:
@@ -16,7 +16,7 @@ python:
   - "3.5"
 cache:
   directories:
-    - $HOME/.pip-cache/
+    - $HOME/.cache/pip/
 install:
   - pip install -U pip
   - pip install --pre --no-binary rasterio rasterio==1.0a1

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,7 @@ env:
 addons:
   apt:
     packages:
+      - libgdal-dev
       - libgdal1h
       - gdal-bin
 python:

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,7 +18,8 @@ cache:
     - $HOME/.pip-cache/
 install:
   - "pip install -U pip"
-  - "pip install numpy>=1.10 -vv"
+  - "pip list"
+  - "pip install numpy>=1.10"
   - "pip install --pre rasterio==1.0a1"
   - "pip list"
   - "pip install -e .[test]"

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,3 +1,7 @@
+1.3a1 (2016-09-27)
+------------------
+- Require rasterio>=1.0a1
+
 1.2.0 (2015-08-03)
 ------------------
 - Register rio-mbtiles to rasterio.rio_plugins (#7).

--- a/mbtiles/__init__.py
+++ b/mbtiles/__init__.py
@@ -9,7 +9,7 @@ from rasterio._io import virtual_file_to_buffer
 
 buffer = bytes if sys.version_info > (3,) else buffer
 
-__version__ = '1.2.0'
+__version__ = '1.3a1'
 
 base_kwds = None
 src = None

--- a/mbtiles/__init__.py
+++ b/mbtiles/__init__.py
@@ -40,6 +40,4 @@ def process_tile(tile):
                 rasterio.band(src, bidx),
                 rasterio.band(tmp, bidx))
 
-    # Get contents of the virtual file and repair it.
-    contents = bytearray(virtual_file_to_buffer('/vsimem/tileimg'))
-    return tile, contents[-1:] + contents[:-1]
+    return tile, bytearray(virtual_file_to_buffer('/vsimem/tileimg'))

--- a/mbtiles/__init__.py
+++ b/mbtiles/__init__.py
@@ -22,7 +22,18 @@ def init_worker(path, profile):
 
 
 def process_tile(tile):
-    """Process a single MBTiles tile."""
+    """Process a single MBTiles tile
+    
+    Parameters
+    ----------
+    tile : mercantile.Tile
+
+    Returns:
+    tile : mercantile.Tile
+        The input tile.
+    bytes : bytearray
+        Image bytes corresponding to the tile.
+    """
     global base_kwds, src
     # Get the bounds of the tile.
     ulx, uly = mercantile.xy(
@@ -34,10 +45,13 @@ def process_tile(tile):
     kwds['transform'] = from_bounds(ulx, lry, lrx, uly, 256, 256)
 
     with rasterio.open('/vsimem/tileimg', 'w', **kwds) as tmp:
-        # Reproject the src dataset into image tile.
-        for bidx in tmp.indexes:
-            reproject(
-                rasterio.band(src, bidx),
-                rasterio.band(tmp, bidx))
+        reproject(rasterio.band(src, src.indexes),
+                  rasterio.band(tmp, tmp.indexes))
 
-    return tile, bytearray(virtual_file_to_buffer('/vsimem/tileimg'))
+    data = bytearray(virtual_file_to_buffer('/vsimem/tileimg'))
+
+    # Workaround for https://bugs.python.org/issue23349.
+    if sys.version_info[0] == 2 and sys.version_info[2] < 10:
+        data[:] = data[-1:] + data[:-1]
+
+    return tile, data

--- a/mbtiles/scripts/cli.py
+++ b/mbtiles/scripts/cli.py
@@ -10,6 +10,7 @@ import click
 import mercantile
 import rasterio
 from rasterio.rio.helpers import resolve_inout
+from rasterio.rio.options import force_overwrite_opt, output_opt
 from rasterio.warp import transform
 
 from mbtiles import buffer, init_worker, process_tile
@@ -26,12 +27,8 @@ DEFAULT_NUM_WORKERS = cpu_count() - 1
     type=click.Path(resolve_path=True),
     required=True,
     metavar="INPUT [OUTPUT]")
-@click.option(
-    '-o', '--output', 'output_opt',
-    default=None,
-    type=click.Path(resolve_path=True),
-    help="Path to output file (optional alternative to a positional arg "
-         "for some commands).")
+@output_opt
+@force_overwrite_opt
 @click.option('--title', help="MBTiles dataset title.")
 @click.option('--description', help="MBTiles dataset description.")
 @click.option('--overlay', 'layer_type', flag_value='overlay', default=True,
@@ -57,8 +54,8 @@ DEFAULT_NUM_WORKERS = cpu_count() - 1
                   DEFAULT_NUM_WORKERS))
 @click.version_option(version=mbtiles_version, message='%(version)s')
 @click.pass_context
-def mbtiles(ctx, files, output_opt, title, description, layer_type,
-            img_format, zoom_levels, image_dump, num_workers):
+def mbtiles(ctx, files, output, force_overwrite, title, description,
+            layer_type, img_format, zoom_levels, image_dump, num_workers):
     """Export a dataset to MBTiles (version 1.1) in a SQLite file.
 
     The input dataset may have any coordinate reference system. It must
@@ -76,14 +73,13 @@ def mbtiles(ctx, files, output_opt, title, description, layer_type,
 
     Python package: rio-mbtiles (https://github.com/mapbox/rio-mbtiles).
     """
-
-    verbosity = (ctx.obj and ctx.obj.get('verbosity')) or 1
-    logger = logging.getLogger('rio')
-
-    output, files = resolve_inout(files=files, output=output_opt)
+    output, files = resolve_inout(files=files, output=output,
+                                  force_overwrite=force_overwrite)
     inputfile = files[0]
 
-    with rasterio.drivers(CPL_DEBUG=verbosity > 2):
+    logger = logging.getLogger('rio-mbtiles')
+
+    with ctx.obj['env']:
 
         # Read metadata from the source dataset.
         with rasterio.open(inputfile) as src:
@@ -100,10 +96,11 @@ def mbtiles(ctx, files, output_opt, title, description, layer_type,
         if zoom_levels:
             minzoom, maxzoom = map(int, zoom_levels.split('..'))
         else:
-            zw = int(round(math.log(360.0/(east-west), 2.0)))
-            zh = int(round(math.log(170.1022/(north-south), 2.0)))
+            zw = int(round(math.log(360.0 / (east - west), 2.0)))
+            zh = int(round(math.log(170.1022 / (north - south), 2.0)))
             minzoom = min(zw, zh)
             maxzoom = max(zw, zh)
+
         logger.debug("Zoom range: %d..%d", minzoom, maxzoom)
 
         # Parameters for creation of tile images.
@@ -157,14 +154,14 @@ def mbtiles(ctx, files, output_opt, title, description, layer_type,
 
         # Constrain bounds.
         EPS = 1.0e-10
-        west = max(-180+EPS, west)
+        west = max(-180 + EPS, west)
         south = max(-85.051129, south)
-        east = min(180-EPS, east)
+        east = min(180 - EPS, east)
         north = min(85.051129, north)
 
         # Initialize iterator over output tiles.
         tiles = mercantile.tiles(
-            west, south, east, north, range(minzoom, maxzoom+1))
+            west, south, east, north, range(minzoom, maxzoom + 1))
 
         for tile, contents in pool.imap_unordered(process_tile, tiles):
 

--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ setup(name='rio-mbtiles',
       install_requires=[
           'click',
           'mercantile',
-          'rasterio>=0.23'
+          'rasterio>=1.0a1'
       ],
       extras_require={
           'test': ['coveralls', 'pytest', 'pytest-cov'],

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -6,6 +6,7 @@ import sys
 from click.testing import CliRunner
 import pytest
 import rasterio
+from rasterio.rio.main import main_group
 
 from mbtiles.scripts.cli import mbtiles
 
@@ -15,16 +16,16 @@ logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
 
 def test_cli_help():
     runner = CliRunner()
-    result = runner.invoke(mbtiles, ['--help'])
+    result = runner.invoke(main_group, ['mbtiles', '--help'])
     assert result.exit_code == 0
     assert "Export a dataset to MBTiles (version 1.1)" in result.output
 
 
-def test_export_metadata(data):
+def test_export_metadata(tmpdir, data):
     inputfile = str(data.join('RGB.byte.tif'))
-    outputfile = str(data.join('export.mbtiles'))
+    outputfile = str(tmpdir.join('export.mbtiles'))
     runner = CliRunner()
-    result = runner.invoke(mbtiles, [inputfile, outputfile])
+    result = runner.invoke(main_group, ['mbtiles', inputfile, outputfile])
     assert result.exit_code == 0
     conn = sqlite3.connect(outputfile)
     cur = conn.cursor()
@@ -32,11 +33,11 @@ def test_export_metadata(data):
     assert cur.fetchone()[1] == 'RGB.byte.tif'
 
 
-def test_export_metadata_output_opt(data):
+def test_export_metadata_output_opt(tmpdir, data):
     inputfile = str(data.join('RGB.byte.tif'))
-    outputfile = str(data.join('export.mbtiles'))
+    outputfile = str(tmpdir.join('export.mbtiles'))
     runner = CliRunner()
-    result = runner.invoke(mbtiles, [inputfile, '-o', outputfile])
+    result = runner.invoke(main_group, ['mbtiles', inputfile, '-o', outputfile])
     assert result.exit_code == 0
     conn = sqlite3.connect(outputfile)
     cur = conn.cursor()
@@ -44,11 +45,11 @@ def test_export_metadata_output_opt(data):
     assert cur.fetchone()[1] == 'RGB.byte.tif'
 
 
-def test_export_tiles(data):
+def test_export_tiles(tmpdir, data):
     inputfile = str(data.join('RGB.byte.tif'))
-    outputfile = str(data.join('export.mbtiles'))
+    outputfile = str(tmpdir.join('export.mbtiles'))
     runner = CliRunner()
-    result = runner.invoke(mbtiles, [inputfile, outputfile])
+    result = runner.invoke(main_group, ['mbtiles', inputfile, outputfile])
     assert result.exit_code == 0
     conn = sqlite3.connect(outputfile)
     cur = conn.cursor()
@@ -56,25 +57,13 @@ def test_export_tiles(data):
     assert len(cur.fetchall()) == 6
 
 
-def test_export_zoom(data):
+def test_export_zoom(tmpdir, data):
     inputfile = str(data.join('RGB.byte.tif'))
-    outputfile = str(data.join('export.mbtiles'))
-    runner = CliRunner()
-    result = runner.invoke(
-        mbtiles, [inputfile, outputfile, '--zoom-levels', '6..7'])
-    assert result.exit_code == 0
-    conn = sqlite3.connect(outputfile)
-    cur = conn.cursor()
-    cur.execute("select * from tiles")
-    assert len(cur.fetchall()) == 6
-
-
-def test_export_jobs(data):
-    inputfile = str(data.join('RGB.byte.tif'))
-    outputfile = str(data.join('export.mbtiles'))
+    outputfile = str(tmpdir.join('export.mbtiles'))
     runner = CliRunner()
     result = runner.invoke(
-        mbtiles, [inputfile, outputfile, '-j', '4'])
+        main_group,
+        ['mbtiles', inputfile, outputfile, '--zoom-levels', '6..7'])
     assert result.exit_code == 0
     conn = sqlite3.connect(outputfile)
     cur = conn.cursor()
@@ -82,12 +71,26 @@ def test_export_jobs(data):
     assert len(cur.fetchall()) == 6
 
 
-def test_export_dump(data):
+def test_export_jobs(tmpdir, data):
     inputfile = str(data.join('RGB.byte.tif'))
-    outputfile = str(data.join('export.mbtiles'))
+    outputfile = str(tmpdir.join('export.mbtiles'))
+    runner = CliRunner()
+    result = runner.invoke(
+        main_group, ['mbtiles', inputfile, outputfile, '-j', '4'])
+    assert result.exit_code == 0
+    conn = sqlite3.connect(outputfile)
+    cur = conn.cursor()
+    cur.execute("select * from tiles")
+    assert len(cur.fetchall()) == 6
+
+
+def test_export_dump(tmpdir, data):
+    inputfile = str(data.join('RGB.byte.tif'))
+    outputfile = str(tmpdir.join('export.mbtiles'))
     dumpdir = pytest.ensuretemp('dump')
     runner = CliRunner()
     result = runner.invoke(
-        mbtiles, [inputfile, outputfile, '--image-dump', str(dumpdir)])
+        main_group,
+        ['mbtiles', inputfile, outputfile, '--image-dump', str(dumpdir)])
     assert result.exit_code == 0
     assert len(os.listdir(str(dumpdir))) == 6

